### PR TITLE
[Docs] Add option to log messages in JSON

### PR DESCRIPTION
**Pull request by dedemorton**
Also added the `logging` namespace to (mostly) match what users see in the config file. There are a few places where we use indentation, but I think it's better to show the full path so there is no ambiguity.

This adds docs for PR #4523

### CODE REVIEW

File: libbeat/docs/loggingconfig.asciidoc [hunk 1]
```diff
@@ -107,37 +107,39 @@ metrics and for this reason they are also not documented.
...
 
 The number of most recent rotated log files to keep on disk. Older files are
 deleted during log rotation. The default value is 7. The `keepfiles` options has to be
 in the range of 2 to 1024 files.
 
 [float]
-==== `files.permissions`
+==== `logging.files.permissions`
+
+//REVIEWERS: Noticed that his is not in the beat.reference.yml file. Should it be? Or is the option old?
```
Comment (andrewkroh) [inline libbeat/docs/loggingconfig.asciidoc]: Yeah, it should be in there too. Can you please add it to [config.reference.yml](https://github.com/elastic/beats/blob/e1060ff722e8f68691ed646599dda575945e5efa/libbeat/_meta/config.reference.yml#L736-L749) and run `make update` from the top-level?

Comment (dedemorton) [reply libbeat/docs/loggingconfig.asciidoc]: @andrewkroh Added. Can you review the changes? Thanks!

File: auditbeat/auditbeat.reference.yml [hunk 2]
```diff
@@ -790,5 +790,9 @@ logging.files:
   # Number of rotated log files to keep. Oldest files will be deleted first.
   #keepfiles: 7
 
+  # The permissions mask to apply when rotating log files. The default value is 0600.
```
Comment (andrewkroh) [inline auditbeat/auditbeat.reference.yml]: This one is different than the others. Perhaps you didn't `git add` the final version of this file before committing?

Comment (dedemorton) [reply auditbeat/auditbeat.reference.yml]: @andrewkroh Good catch! Weirdness...I committed by using `git commit -a -m ...` so I don't know how it's possible that the files weren't updated/committed correctly. Ah well...it's fixed now.

**Pull request merged**